### PR TITLE
Fix compiler warnings in wasm2c output

### PR DESCRIPTION
**Pull request by sbc100**
Use stricter compiler settings and fix all the resulting warnings.
This is precursor to landing windows compiler support for wasm2c.
See #1843.

### CODE REVIEW

File: src/c-writer.cc [hunk 1]
```diff
@@ -1017,8 +1023,15 @@ void CWriter::WriteMemory(const std::string& name) {
 }
 
 void CWriter::WriteTables() {
-  if (module_->tables.size() == module_->num_table_imports)
+  if (module_->tables.size() == module_->num_table_imports) {
     return;
+  }
+
+  if (!module_->types.size()) {
+    // If not types are defined then there is no way to use the table
```
Comment (kripken) [inline src/c-writer.cc:1031]:
```suggestion
    // If no types are defined then there is no way to use the table
```

File: src/c-writer.cc [hunk 2]
```diff
@@ -1082,17 +1100,27 @@ void CWriter::WriteDataInitializers() {
...
   }
 
   Write(CloseBrace(), Newline());
 }
 
 void CWriter::WriteElemInitializers() {
+  Write(Newline(), "static void init_table(void) ", OpenBrace());
+
+  if (!module_->types.size()) {
+    // If there are no function types there cannot be any table entries either.
```
Comment (kripken) [inline src/c-writer.cc:1115]: I don't think that's true in general. You can have a table that contains non-function types with wasm GC at least.

**Review by kripken — APPROVED** [review]
lgtm % comments

File: src/c-writer.cc (hunk 2 above)
Comment (sbc100) [reply src/c-writer.cc:1115]: Right, but wouldn't the mean that `module_->types` would have to be non-empty?  I'll update the comment.

Comment (kripken) [reply src/c-writer.cc]: Ah, yes, if these are all types and not just function types, which I realize now must be the case, then just updating the comment sgtm.

**Pull request merged**